### PR TITLE
Chore/tag npm publish

### PR DESCRIPTION
**Pull request by seromenho**
- CI now publish to npm once we create a release tag
- Fixed exit codes when login or set org fails

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
     - fliplet create-widget awesome-widget
-    - cd awesome-widget && fliplet test
\ No newline at end of file
+    - cd awesome-widget && fliplet test
+
+deployment:
+  npm:
+    tag: /v[0-9]+(\.[0-9]+)*/
+    commands:
+      - npm publish
```
Comment (squallstar) [inline circle.yml]: pretty sure you can't publish without logging as a authenticated user

File: circle.yml [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
     - fliplet env staging
     - fliplet login -u $TEST_USER -p $TEST_PASSWORD
     - fliplet organization $TEST_ORGANIZATION
     - fliplet create-widget awesome-widget
-    - cd awesome-widget && fliplet test
\ No newline at end of file
+    - cd awesome-widget && fliplet test
+
+deployment:
+  npm:
```
Comment (squallstar) [inline circle.yml]: why not master?

Comment (seromenho) [inline circle.yml]: I think we should tag releases so people can historically see what we are fixing or adding as features.
Other projects use tag releases almost like a changelog.
What do you think?

File: circle.yml [hunk 3]
```diff
@@ -1,11 +1,18 @@
 dependencies:
   pre:
     - npm link
+    - echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
```
Comment (squallstar) [inline circle.yml]: nice one

File: circle.yml (hunk 1 above)
Comment (seromenho) [inline circle.yml]: The publish reads an auth token from `~/.npmrc`

File: circle.yml (hunk 2 above)
Comment (squallstar) [inline circle.yml]: makes sense, please add the steps to create a release on the readme so we know how to publish to npm.

File: circle.yml (hunk 1 above)
Comment (squallstar) [inline circle.yml]: yeah sorry seen that

File: circle.yml (hunk 2 above)
Comment (seromenho) [inline circle.yml]: Added.

**Comment by seromenho** [comment]
@squallstar ready to review again. Ty

**Pull request merged**